contracts/Archetype.sol

```diff
@@ -112,24 +112,28 @@ contract Archetype is
     address affiliate,
     bytes calldata signature
   ) external payable {
-    if (quantityList.length != toList.length) {
-      revert InvalidConfig();
-    }
+    require(quantityList.length == toList.length, "InvalidConfig");
 
     DutchInvite storage invite = invites[auth.key];
-    uint256 quantity = 0;
-    {
-      uint32 unitSize = invite.unitSize;
-      for (uint256 i = 0; i < quantityList.length; i++) {
-        if (unitSize > 1) {
-          quantity += quantityList[i] * unitSize;
-        } else {
-          quantity += quantityList[i];
-        }
+    uint256 curSupply = _totalMinted();
+    uint256 quantity;
+
+    for (uint256 i; i < toList.length; ) {
+      uint256 quantityToAdd;
+      if (invite.unitSize > 1) {
```

Comment:
Is this cheaper than loading invite.unitSize into a mem var? Or is it about the same.